query/src/main/java/org/infinispan/query/dsl/embedded/impl/EmbeddedLuceneQuery.java

```diff
@@ -83,21 +83,23 @@ public void resetQuery() {
 
    @Override
    public QueryResult<T> execute() {
-      List<T> results = StreamSupport.stream(spliterator(), false).collect(Collectors.toList());
-      long hits = createCacheQuery().getResultSize();
-      return new QueryResultImpl<>(hits, results);
+      List<?> results = StreamSupport.stream(spliterator(), false)
+            .map(i -> (projection == null) ? i : convertProjectionItem(i))
+            .collect(Collectors.toList());
+
+      int hits = Math.toIntExact(createCacheQuery().getResultSize());
+      return new QueryResultImpl<>(hits, (List<T>) results);
    }
 
    @Override
    public CloseableIterator<T> iterator() {
-      CloseableIterator<T> iterator = createCacheQuery().iterator();
-      return rowProcessor == null ? iterator : new MappingIterator(iterator, t -> rowProcessor.apply((Object[]) t));
+      IndexedQuery<T> cacheQuery = createCacheQuery();
+      return new MappingIterator(cacheQuery.iterator(), i -> (projection == null) ? i : convertProjectionItem(i) );
    }
 
    @Override
    public int getResultSize() {
-      //todo [anistor] optimize this by running a slightly modified query that performs just COUNT only, ignoring projections or sorting
```

Comment:
Did you fix this @fax4ever ? If not, let's keep it as a reminder :)

Reply:
I have to check :P

Reply:
From what I can see it seems that we did it. Because `fetchTotalHitCount` Search 6 API invokes a specific work that does not involve pagination and sorting. @yrodiere Did I say something wrong? (Thanks)

Reply:
You're right. And from what I can see `getResultSize()` calls `fetchTotalHitCount`, so we already optimize everything we need.